### PR TITLE
PDI 19373 Avro Output Step: Generates corrupt Avro file when connecte…

### PR DESCRIPTION
**Pull request by GopiKrishnaAngadi**
…d to repository from Spoon

### CODE REVIEW

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 2m 30s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl kettle-plugins/formats-meta
```
---
 :exclamation: No tests found!
 ---
 
 

:information_source:  _This is an automatic message_

File: kettle-plugins/formats-meta/src/main/java/org/pentaho/big/data/kettle/plugins/formats/avro/output/AvroOutputMetaBase.java [hunk 1]
```diff
@@ -286,7 +286,12 @@ public void saveRep( Repository rep, IMetaStore metaStore, ObjectId transformati
         rep.saveStepAttribute( transformationId, stepId, i, PRECISION, field.getPrecision() );
         rep.saveStepAttribute( transformationId, stepId, i, SCALE, field.getScale() );
         rep.saveStepAttribute( transformationId, stepId, i, NULLABLE, Boolean.toString( field.getAllowNull() ) );
-        rep.saveStepAttribute( transformationId, stepId, i, DEFAULT, field.getDefaultValue() );
+        String defaultValue = field.getDefaultValue();
```
Comment (ricksjames) [inline kettle-plugins/formats-meta/src/main/java/org/pentaho/big/data/kettle/plugins/formats/avro/output/AvroOutputMetaBase.java:289]: @GopiKrishnaAngadi Please review and fix the checkstyle findings in sonar.

File: kettle-plugins/formats-meta/src/main/java/org/pentaho/big/data/kettle/plugins/formats/avro/output/AvroOutputMetaBase.java [hunk 2]
```diff
@@ -286,7 +286,12 @@ public void saveRep( Repository rep, IMetaStore metaStore, ObjectId transformati
...
+        int length = defaultValue.length();
+        if(length>0)
+          rep.saveStepAttribute( transformationId, stepId, i, DEFAULT, defaultValue );
+        else
+          rep.saveStepAttribute( transformationId, stepId, i, DEFAULT, null );
```
Comment (ricksjames) [inline kettle-plugins/formats-meta/src/main/java/org/pentaho/big/data/kettle/plugins/formats/avro/output/AvroOutputMetaBase.java:294]: I'd like to see this written in a way where the rep.saveStepAttribute(...) method did not have to be repeated. Something like this:
        String defaultValue = StringUtil.isEmpty( field.getDefaultValue() ) ? null : field.getDefaultValue();
        rep.saveStepAttribute( transformationId, stepId, i, DEFAULT, defaultValue );

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 3m 5s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl kettle-plugins/formats-meta
```
---
 :exclamation: No tests found!
 ---
 
 

:information_source:  _This is an automatic message_

**Pull request merged**